galaxy_importer/utils/execution_environment.py

```diff
@@ -0,0 +1,116 @@
+# (c) 2012-2019, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import os
+import re
+
+from galaxy_importer.utils import yaml as yaml_utils
+
+
+def process_execution_environment(path, logger):
+    ex_env = {}
+    ex_env_path = os.path.join(path, 'meta', 'execution-environment.yml')
+
+    if os.path.exists(ex_env_path):
+        ex_env = _load_yaml(ex_env_path, logger)
+    else:
+        logger.info('No execution environment data found.')
+        return ex_env
+
+    if 'dependencies' in ex_env and 'galaxy' in ex_env['dependencies'] and os.path.exists(
+        os.path.join(path, ex_env['dependencies']['galaxy'])
+    ):
+        logger.info('Linting collection dependencies')
+        galaxy_contents = _load_yaml(
+            os.path.join(path, ex_env['dependencies']['galaxy']),
+            logger
+        )
+        logger.info('Loading collection dependencies')
+        ex_env = _write_to_ee(ex_env, 'galaxy', galaxy_contents)
+    else:
+        logger.warning('Galaxy dependencies file not found')
```

Comment:
If we always expect there to be sections for `galaxy`, `python`, `system` and therefore files, then these warnings make sense. If it is OK that a user doesn't enter a `galaxy` section for example, we still warn them `file not found` - I think that could be confusing.

If a missing dependencies key should throw a warning, could check and exit early:
```
if 'dependencies' not in ex_env:
    logger.warning('No dependencies found')
    return ex_env
```

Reply:
Only python and system entries here make sense. The spec allows the _user's definition_ to give a Galaxy requirements file. On the collection level, dependencies are specified in `galaxy.yml`, not here. If anything, you should error if you find a `galaxy` entry here.